### PR TITLE
adding random delay between pokemon capture & transfer. #774

### PR DESCRIPTION
**Pull request by SpaceWhale**
Short Description: 
random delay between capture & transfer.

### CODE REVIEW

File: pokemongo_bot/cell_workers/pokemon_catch_worker.py [hunk 1]
```diff
@@ -216,6 +216,9 @@ def work(self):
                                         if len(pokemon_to_transfer) == 0:
                                             raise RuntimeError(
                                                 'Trying to transfer 0 pokemons!')
+                                        # Add slight delay between capture & candy transfer #774
+                                        logger.log("Waiting briefly before transferring pokemon")
+                                        sleep(3)
```
Comment (azack) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: @SpaceWhale What's random about 3?

Comment (VirtualSatai) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: Maybe he used a method like this: https://xkcd.com/221/

Comment (xc-mezcal) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: Maybe a random 1-3 second? not exactly 3?

Comment (xc-mezcal) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: should random 1-3 not 3 :D

Comment (cwild) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: To be fair, the sleep implementation allows for randomness..
https://github.com/PokemonGoF/PokemonGo-Bot/blob/9ea015e1fd9c5d269276de8a8e822ee8eb0afa0f/pokemongo_bot/human_behaviour.py#L8-L11

Comment (SpaceWhale) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: the sleep implementation does allow for randomness but I appear to be using it wrong. I'm going to add little function and update the pull request.

Comment (SpaceWhale) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: This isn't `time.sleep(3)` it's `sleep(3)` using the function in human_behavior.py. 10 sample outputs using this function as implemented:

>2.67440283374
>2.43061394227
>3.41717271461
>3.16960045279
>3.60687241192
>3.77872126265
>2.84796236007
>2.64138294609
>3.11884116248
>2.61486735064

Comment (SpaceWhale) [inline pokemongo_bot/cell_workers/pokemon_catch_worker.py]: I added a click_action_delay function where a range can be set instead.

**Comment by bstpierr** [comment]
As with #1205, this should be customizable within the settings for people who want lightning fast botting.

**Comment by fierysolid** [comment]
@douglascamata this makes the bot more humanlike, can we get a couple approvals? Looks good to me.

File: pokemongo_bot/human_behaviour.py [hunk 2]
```diff
@@ -11,8 +11,14 @@ def jitter(value, delta=0.3):
...
+    # Waits for random number of seconds between low & high numbers
+    longNum = uniform(low, high)
+    shortNum = float("{0:.2f}".format(longNum))
+    time.sleep(shortNum)
 
 def random_lat_long_delta():
     # Return random value from [-.000025, .000025]. Since 364,000 feet is equivalent to one degree of latitude, this
     # should be 364,000 * .000025 = 9.1. So it returns between [-9.1, 9.1]
     return ((random() * 0.00001) - 0.000005) * 5
+click_action_delay(1, 3)
```
Comment (cwild) [inline pokemongo_bot/human_behaviour.py]: Is this intentional? It will block when the module is imported.

**Comment by eggins** [comment]
:+1:  has my approval

**Comment by eggins** [comment]
:+1:

**Comment by TheSavior** [comment]
This seems kinda silly. There is no way that *this* is going to be the way niantic uses to catch people botting. I feel like this will just slow down the bot for no gain. :-/

**Comment by SpaceWhale** [comment]
@TheSavior tracking request frequency is definitely a method to detect bots. You can virtually disable this in the configs if you _feel_ this is a waste.

**Pull request merged**